"panic: assignment to entry in nil map" when using `gopass setup` with keychain enabled

### Summary

I tried creating new password store with age backend and keychain enabled, but I'm getting following error during the process: "panic: assignment to entry in nil map".

### Steps To Reproduce

```sh
export GOPASS_DEBUG=1
export GOPASS_DEBUG_LOG=gopass.log
gopass config keychain true
gopass setup --crypto age --storage fs
```

### Expected behavior

New password store should be successfully created and keychain should be used without any issues.

### Environment

- OS: Tails 5.4
- OS version: `Linux amnesia 5.10.0-17-amd64 #1 SMP Debian 5.10.136-1 (2022-08-13) x86_64 GNU/Linux`
- gopass Version: `gopass 1.14.8 go1.19.1 linux amd64`
- Installation method: GitHub Releases ([gopass_1.14.9_linux_amd64.deb](https://github.com/gopasspw/gopass/releases/download/v1.14.9/gopass_1.14.9_linux_amd64.deb))

### Additional context

`gopass config keychain true`

```sh
true
```

`gopass setup --crypto age --storage fs`

```sh

   __     _    _ _      _ _   ___   ___
 /'_ '\ /'_'\ ( '_'\  /'_' )/',__)/',__)
( (_) |( (_) )| (_) )( (_| |\__, \\__, \
'\__  |'\___/'| ,__/''\__,_)(____/(____/
( )_) |       | |
 \___/'       (_)

🌟 Welcome to gopass!
🌟 Initializing a new password store ...
🔐 No useable cryptographic keys. Generating new key pair
🧪 Creating cryptographic key pair (age) ...
⚠ Do you want to enter a passphrase? (otherwise we generate one for you) [y/N/q]: 
✅ Key pair generated
Passphrase: gains cable defiance ouch
⚠ You need to remember this very well!
⚠ 🔐 We need to unlock your newly created private key now! Please enter the passphrase you just generated.
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/gopasspw/gopass/internal/backend/crypto/age.(*osKeyring).Set(0xc000294038, {0xc000258780, 0x2b}, {0xc00002a940?, 0x1?})
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/askpass.go:44 +0x14e
github.com/gopasspw/gopass/internal/backend/crypto/age.(*askPass).Passphrase(0xc0000108d0, {0xc000258780, 0x2b}, {0xc000292280, 0x48}, 0x18?)
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/askpass.go:104 +0x2c3
github.com/gopasspw/gopass/internal/backend/crypto/age.(*Age).Identities.func1({0xc000258780, 0x2b}, 0x15?)
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/identities.go:27 +0x9d
github.com/gopasspw/gopass/internal/backend/crypto/age.(*Age).decryptFile(0xc000258780?, {0xcecf30, 0xc00028a180}, {0xc000258780, 0x2b})
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/decrypt.go:58 +0x163
github.com/gopasspw/gopass/internal/backend/crypto/age.(*Age).Identities(0xc00028b170, {0xcecf30, 0xc00028a120})
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/identities.go:35 +0x22c
github.com/gopasspw/gopass/internal/backend/crypto/age.(*Age).getNativeIdentities(0x54e8f6?, {0xcecf30?, 0xc00028a120?})
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/identities.go:291 +0x25
github.com/gopasspw/gopass/internal/backend/crypto/age.(*Age).getAllIdentities(0xc0003f5720?, {0xcecf30, 0xc00028a120})
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/identities.go:224 +0x85
github.com/gopasspw/gopass/internal/backend/crypto/age.(*Age).ListIdentities(0xcecf30?, {0xcecf30, 0xc00028a120})
	/home/runner/work/gopass/gopass/internal/backend/crypto/age/identities.go:102 +0x85
github.com/gopasspw/gopass/internal/action.(*Action).initGenerateIdentity(0xcecf30?, {0xcecf30, 0xc0004fa630}, {0xcf2c60, 0xc00028b170}, {0xc000026075, 0x7}, {0x0, 0x0})
	/home/runner/work/gopass/gopass/internal/action/setup.go:172 +0x642
github.com/gopasspw/gopass/internal/action.(*Action).initCheckPrivateKeys(0xbb98c2?, {0xcecf30, 0xc0004fa630}, {0xcf2c60, 0xc00028b170})
	/home/runner/work/gopass/gopass/internal/action/setup.go:104 +0x18f
github.com/gopasspw/gopass/internal/action.(*Action).Setup(0xc0000efb20, 0xc000088e80?)
	/home/runner/work/gopass/gopass/internal/action/setup.go:79 +0x635
github.com/urfave/cli/v2.(*Command).Run(0xc00023aa20, 0xc000088b80)
	pkg/mod/github.com/urfave/cli/v2@v2.16.3/command.go:173 +0x6ca
github.com/urfave/cli/v2.(*App).RunContext(0xc00016c380, {0xcecf30?, 0xc0003ed110}, {0xc0000221e0, 0x6, 0x6})
	pkg/mod/github.com/urfave/cli/v2@v2.16.3/app.go:383 +0xfde
main.main()
	/home/runner/work/gopass/gopass/main.go:89 +0x452
```

`env`

```sh
SHELL=/bin/bash
SESSION_MANAGER=local/amnesia:@/tmp/.ICE-unix/8064,unix/amnesia:/tmp/.ICE-unix/8064
SOCKS5_SERVER=127.0.0.1:9050
QT_ACCESSIBILITY=1
COLORTERM=truecolor
HISTCONTROL=ignoreboth
XDG_MENU_PREFIX=gnome-
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
XMODIFIERS=@im=ibus
ONIONSHARE_HIDE_TOR_SETTINGS=1SHELL=/bin/bash
SESSION_MANAGER=local/amnesia:@/tmp/.ICE-unix/8064,unix/amnesia:/tmp/.ICE-unix/8064
SOCKS5_SERVER=127.0.0.1:9050
QT_ACCESSIBILITY=1
COLORTERM=truecolor
HISTCONTROL=ignoreboth
XDG_MENU_PREFIX=gnome-
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
XMODIFIERS=@im=ibus
ONIONSHARE_HIDE_TOR_SETTINGS=1
DESKTOP_SESSION=gnome-xorg
GTK_MODULES=gail:atk-bridge
DBUS_STARTER_BUS_TYPE=session
PWD=/home/amnesia/Tor Browser/gopass_1.14.8_linux_amd64/data/usr/bin
XDG_SESSION_DESKTOP=gnome-xorg
LOGNAME=amnesia
XDG_SESSION_TYPE=x11
GOPASS_DEBUG=1
GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1
XAUTHORITY=/run/user/1000/gdm/Xauthority
TAILS_WIKI_SUPPORTED_LANGUAGES=en de es fr it pt ru
QT_STYLE_OVERRIDE=adwaita
WINDOWPATH=2
GDM_LANG=en_US.UTF-8
HOME=/home/amnesia
USERNAME=amnesia
LANG=en_US.UTF-8
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
XDG_CURRENT_DESKTOP=GNOME
VTE_VERSION=6203
GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/f340c1b5_3aeb_4d17_8614_47eb52b51f57
SOCKS_SERVER=127.0.0.1:9050
GOPASS_DEBUG_LOG=gopass.log
LESSCLOSE=/usr/bin/lesspipe %s %s
XDG_SESSION_CLASS=user
TERM=xterm-256color
LESSOPEN=| /usr/bin/lesspipe %s
USER=amnesia
GNOME_TERMINAL_SERVICE=:1.99
DISPLAY=:1
SHLVL=1
INSIDE_NAUTILUS_PYTHON=
QT_IM_MODULE=ibus
DBUS_STARTER_ADDRESS=unix:path=/run/user/1000/bus,guid=9cd2a38b7260d8ad0f61d1f66334023a
XDG_RUNTIME_DIR=/run/user/1000
NODE_PATH=/usr/local/lib/nodejs
XDG_DATA_DIRS=/usr/share/gnome:/usr/local/share/:/usr/share/
PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
GDMSESSION=gnome-xorg
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus,guid=9cd2a38b7260d8ad0f61d1f66334023a
_=/usr/bin/env
```

I'm not sure if it's relevant, but I have `gnome-keyring 3.36.0-1` installed. I also attached 
[gopass.log](https://github.com/gopasspw/gopass/files/9664112/gopass.log) file that was generated during the process. Let me know if you need any additional informations.

I hope it will be fixed by #2356 but I opened this issue in case if any other users would experience this.

--- a/internal/backend/crypto/age/askpass.go
+++ b/internal/backend/crypto/age/askpass.go
@@ -25,15 +25,18 @@ type osKeyring struct {
 	knownKeys map[string]bool
 }
 
+func newOsKeyring() *osKeyring {
+	return &osKeyring{
+		knownKeys: make(map[string]bool),
+	}
+}
+
 func (o *osKeyring) Get(key string) (string, bool) {
 	sec, err := keyring.Get("gopass", key)
 	if err != nil {
 		debug.Log("failed to get %s from OS keyring: %w", key, err)
 
 		return "", false
-	}
-	if o.knownKeys == nil {
-		o.knownKeys = make(map[string]bool, 1)
 	}
 	o.knownKeys[name] = true
 
@@ -44,9 +47,6 @@ func (o *osKeyring) Set(name, value string) {
 	if err := keyring.Set("gopass", name, value); err != nil {
 		debug.Log("failed to set %s: %w", name, err)
 	}
-	if o.knownKeys == nil {
-		o.knownKeys = make(map[string]bool, 1)
-	}
 	o.knownKeys[name] = true
 }
 
@@ -55,9 +55,6 @@ func (o *osKeyring) Remove(name string) {
 		debug.Log("failed to remove %s from keyring: %s", name, err)
 
 		return
-	}
-	if o.knownKeys == nil {
-		o.knownKeys = make(map[string]bool, 1)
 	}
 	o.knownKeys[name] = false
 }
@@ -86,7 +83,7 @@ func newAskPass(ctx context.Context) *askPass {
 	}
 
 	if err := keyring.Set("gopass", "sentinel", "empty"); err == nil && IsUseKeychain(ctx) {
-		a.cache = &osKeyring{}
+		a.cache = newOsKeyring()
 	}
 
 	return a

--- a/internal/backend/crypto/age/askpass.go
+++ b/internal/backend/crypto/age/askpass.go
@@ -83,6 +83,7 @@ func newAskPass(ctx context.Context) *askPass {
 	}
 
 	if err := keyring.Set("gopass", "sentinel", "empty"); err == nil && IsUseKeychain(ctx) {
+		debug.Log("using OS keychain to cache age credentials")
 		a.cache = newOsKeyring()
 	}
 

--- a/internal/backend/crypto/age/askpass.go
+++ b/internal/backend/crypto/age/askpass.go
@@ -32,6 +32,9 @@ func (o *osKeyring) Get(key string) (string, bool) {
 
 		return "", false
 	}
+	if o.knownKeys == nil {
+		o.knownKeys = make(map[string]bool, 1)
+	}
 	o.knownKeys[name] = true
 
 	return sec, true
@@ -41,6 +44,9 @@ func (o *osKeyring) Set(name, value string) {
 	if err := keyring.Set("gopass", name, value); err != nil {
 		debug.Log("failed to set %s: %w", name, err)
 	}
+	if o.knownKeys == nil {
+		o.knownKeys = make(map[string]bool, 1)
+	}
 	o.knownKeys[name] = true
 }
 
@@ -49,6 +55,9 @@ func (o *osKeyring) Remove(name string) {
 		debug.Log("failed to remove %s from keyring: %s", name, err)
 
 		return
+	}
+	if o.knownKeys == nil {
+		o.knownKeys = make(map[string]bool, 1)
 	}
 	o.knownKeys[name] = false
 }

